tests/liteclientHSM_test.cc

```diff
@@ -0,0 +1,1179 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <boost/process.hpp>
+#include <boost/process/env.hpp>
+
+#include "libaktualizr/types.h"
+#include "logging/logging.h"
+#include "test_utils.h"
+#include "uptane_generator/image_repo.h"
+#include "utilities/utils.h"
+
+#include "composeappmanager.h"
+#include "liteclient.h"
+
+#include <iostream>
+#include <string>
+
+#include "helpers.h"
+#include "ostree/repo.h"
+#include "target.h"
+
+using ::testing::NiceMock;
+using ::testing::Return;
+
+static std::string executeCmd(const std::string& cmd, const std::vector<std::string>& args, const std::string& desc) {
+  auto res = Process::spawn(cmd, args);
+  if (std::get<0>(res) != 0) throw std::runtime_error("Failed to " + desc + ": " + std::get<2>(res));
+
+  auto std_out = std::get<1>(res);
+  boost::trim_right_if(std_out, boost::is_any_of(" \t\r\n"));
+  return std_out;
+}
+
+/**
+ * Class MockAppEngine
+ *
+ */
+class MockAppEngine : public AppEngine {
+ public:
+  MockAppEngine(bool default_behaviour = true) {
+    if (!default_behaviour) return;
+
+    ON_CALL(*this, fetch).WillByDefault(Return(true));
+    ON_CALL(*this, install).WillByDefault(Return(true));
+    ON_CALL(*this, run).WillByDefault(Return(true));
+    ON_CALL(*this, isRunning).WillByDefault(Return(true));
+  }
+
+ public:
+  MOCK_METHOD(bool, fetch, (const App& app), (override));
+  MOCK_METHOD(bool, install, (const App& app), (override));
+  MOCK_METHOD(bool, run, (const App& app), (override));
+  MOCK_METHOD(void, remove, (const App& app), (override));
+  MOCK_METHOD(bool, isRunning, (const App& app), (const, override));
+};
+
+/**
+ * Class SysRootFS
+ *
+ */
+class SysRootFS {
```

Comment:
I moved the common test components into the "fixture" directory https://github.com/foundriesio/aktualizr-lite/blob/master/tests/fixtures/liteclienttest.cc so you can make use of them by including them into the given test and remove their implementation from this test.

Reply:
ok